src/main/java/org/commcare/formplayer/aspects/MetricsAspect.java

```diff
@@ -41,46 +48,77 @@
+                if (sessionId != null) {
+                    try {
+                        fetchTimer = new SimpleTimer();
+                        fetchTimer.start();
+                        SerializableFormSession serializableFormSession = formSessionRepo.findOneWrapped(bean.getSessionId());
+                        formName = serializableFormSession.getTitle();
+                        fetchTimer.end();
+                    } catch (FormNotFoundException e) {
+
+                    }
+                }
+            }
         }
 
         SimpleTimer timer = new SimpleTimer();
         timer.start();
         Object result = joinPoint.proceed();
         timer.end();
 
+        List<String> datadogArgs = new ArrayList<>();
+        datadogArgs.add("domain:" + domain);
+        datadogArgs.add("request:" + requestPath);
+        datadogArgs.add("duration:" + timer.getDurationBucket());
+        datadogArgs.add("unblocked_time:" + getUnblockedTimeBucket(timer));
+        datadogArgs.add("blocked_time:" + getBlockedTimeBucket());
+        datadogArgs.add("restore_blocked_time:" + getRestoreBlockedTimeBucket());
+        datadogArgs.add("install_blocked_time:" + getInstallBlockedTimeBucket());
+        datadogArgs.add("submit_blocked_time:" + getSubmitBlockedTimeBucket());
+
+        // optional datadog args
+        if (formName != null) {
+            datadogArgs.add("form_name:" + formName);
+        }
+
         datadogStatsDClient.increment(
                 Constants.DATADOG_REQUESTS,
-                "domain:" + domain,
-                "request:" + requestPath,
-                "duration:" + timer.getDurationBucket(),
-                "unblocked_time:" + getUnblockedTimeBucket(timer),
-                "blocked_time:" + getBlockedTimeBucket(),
-                "restore_blocked_time:" + getRestoreBlockedTimeBucket(),
-                "install_blocked_time:" + getInstallBlockedTimeBucket(),
-                "submit_blocked_time:" + getSubmitBlockedTimeBucket()
+                datadogArgs.toArray(new String[datadogArgs.size()])
         );
 
         datadogStatsDClient.recordExecutionTime(
                 Constants.DATADOG_TIMINGS,
                 timer.durationInMs(),
-                "domain:" + domain,
-                "request:" + requestPath,
-                "duration:" + timer.getDurationBucket(),
-                "unblocked_time:" + getUnblockedTimeBucket(timer),
-                "blocked_time:" + getBlockedTimeBucket(),
-                "restore_blocked_time:" + getRestoreBlockedTimeBucket(),
-                "install_blocked_time:" + getInstallBlockedTimeBucket(),
-                "submit_blocked_time:" + getSubmitBlockedTimeBucket()
+                datadogArgs.toArray(new String[datadogArgs.size()])
         );
+
+        if (fetchTimer != null) {
+            datadogArgs.add("task:fetch_form_session");
+            datadogStatsDClient.recordExecutionTime(
+                    "test_timings",
```

Comment:
@calellowitz does it seem worth adding this tag as a constant? I figure I'll remove this code in the near future and `test_timings` will either never be used again or prove to be somewhat useful.

Reply:
Rather than name it `test_timings` and add a tag for `fetch_form_session`, it might make more sense to just name the metric `fetch_form_session`, unless we expect to aggregate this with other timings

Reply:
Updated 👍